hugo/assets/scripts/domainmodel/domainmodel-tools.ts

```diff
@@ -22,6 +13,24 @@ export function getTreeNode(ast: AstNode): TreeNode {
         }
     }
 
+    // create a TreeNode from a PackageDeclaration
+    const getPackageDeclarationTreeNode = (p: PackageDeclaration): TreeNode => {
+        return {
+            ...p,
+            children: p.elements.map(e => getTreeNode(e))
+        }
+    }
+
+    // create a TreeNode from any DomainModelElement
+    const getTreeNode = (e: DomainModelElement): TreeNode => {
```

Comment:
Please rename the helper method, it has the same name as the parent function. This becomes confusing, because you do a recursive call and it is not clear which one is meant.

Otherwise looks good to me.